tests/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Razor.Tests/RazorSourceUpdaterTests.cs

```diff
@@ -0,0 +1,387 @@
+                            new LocationLookup("RazorUpdaterStepViews\\TestViews\\View.cshtml.cs", "WriteLiteral(\"    <div>\\r\\n        <p>\")"),
+                        },
+                    },
+                    new[]
+                    {
+                        GetFullPath("RazorUpdaterStepViews\\TestViews\\View.cshtml.cs"),
+                    },
+                },
+
+                // Diagnostics in multiple files (from multiple analyzers)
+                new object[]
+                {
+                    new[]
+                    {
+                        new[]
+                        {
+                            new LocationLookup("RazorUpdaterStepViews\\TestViews\\View.cshtml.cs", "using Microsoft.AspNetCore.Mvc;"),
+                            new LocationLookup("RazorUpdaterStepViews\\TestViews\\View.cshtml.cs", "Model[0]"),
+                        },
+                        new[]
+                        {
+                            new LocationLookup("RazorUpdaterStepViews\\TestViews\\View.cshtml.cs", "Model[1]"),
+                            new LocationLookup("RazorUpdaterStepViews\\TestViews\\Simple.cshtml.cs", "using Microsoft.AspNetCore.Mvc;"),
+                            new LocationLookup("RazorUpdaterStepViews\\TestViews\\Simple.cshtml.cs", "DateTime.Now.ToString()"),
+                        }
+                    },
+                    new[]
+                    {
+                        GetFullPath("RazorUpdaterStepViews\\_ViewImports.cshtml"),
+                        GetFullPath("RazorUpdaterStepViews\\TestViews\\View.cshtml"),
+                        GetFullPath("RazorUpdaterStepViews\\TestViews\\Simple.cshtml"),
+                    },
+                }
+            };
+
+        [Fact]
+        public async Task ApplyNegativeTests()
+        {
+            using var mock = GetMock(null, Array.Empty<LocationLookup[]>());
+            var updater = mock.Create<RazorSourceUpdater>();
+
+            await Assert.ThrowsAsync<ArgumentNullException>("context", () => updater.ApplyAsync(null!, ImmutableArray.Create<RazorCodeDocument>(), CancellationToken.None)).ConfigureAwait(true);
+            await Assert.ThrowsAsync<InvalidOperationException>(() => updater.ApplyAsync(mock.Mock<IUpgradeContext>().Object, ImmutableArray.Create<RazorCodeDocument>(), CancellationToken.None)).ConfigureAwait(true);
+        }
+
+        [Theory]
+        [MemberData(nameof(ApplyData))]
+        public async Task ApplyTests(LocationLookup[][] diagnosticLocations, string[] expectedUpdatedFiles, TextReplacement[] expectedReplacements)
+        {
+            // Arrange
+            using var mock = GetMock("RazorUpdaterStepViews/Test.csproj", diagnosticLocations);
+            var razorDocs = await GetRazorCodeDocumentsAsync(mock).ConfigureAwait(true);
+            var context = mock.Mock<IUpgradeContext>();
```

Comment:
nit: `mock.Mock<>()` should only be used when it's for DI. If it's something you're passing into a message, it can cause problems (ie, all calls to `mock.Mock<>()` will return the same instance). Otherwise, you can use `new Mock<>()` to get a separate instance.